internal/auth0/action/resource_trigger_action.go

```diff
@@ -111,6 +118,59 @@ func createTriggerAction(_ context.Context, d *schema.ResourceData, m interface{
 	return nil
 }
 
+func updateTriggerAction(_ context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	api := m.(*config.Config).GetAPI()
+
+	trigger := d.Get("trigger").(string)
+	actionID := d.Get("action_id").(string)
+	displayName := d.Get("display_name").(string)
+
+	currentBindings, err := api.Action.Bindings(trigger)
+	if err != nil {
+		if mErr, ok := err.(management.Error); ok && mErr.Status() == http.StatusNotFound {
+			d.SetId("")
+			return nil
+		}
+		return diag.FromErr(err)
+	}
+
+	found := false
+	var updatedBindings []*management.ActionBinding
+	for _, binding := range currentBindings.Bindings {
+		if binding.Action.GetID() == actionID {
+			d.SetId(trigger + "::" + actionID)
+			updatedBindings = append(updatedBindings, &management.ActionBinding{
+				Ref: &management.ActionBindingReference{
+					Type:  auth0.String("action_id"),
+					Value: &actionID,
+				},
+				DisplayName: &displayName,
+			})
+			found = true
+		} else {
+			updatedBindings = append(updatedBindings, &management.ActionBinding{
+				Ref: &management.ActionBindingReference{
+					Type:  auth0.String("action_id"),
+					Value: binding.Action.ID,
+				},
+				DisplayName: binding.DisplayName,
+			})
+		}
```

Comment:
If you continue in the if above, we don't need an else here so we can reduce nesting.